extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -710,32 +716,34 @@ private void handleHibernateORMWithNoPersistenceXml(
         }
     }
 
-    private Optional<String> guessDialect(Optional<String> driver) {
+    private Optional<String> guessDialect(Optional<String> kind) {
         // For now select the latest dialect from the driver
         // later, we can keep doing that but also avoid DCE
         // of all the dialects we want in so that people can override them
-        String resolvedDriver = driver.orElse("NODRIVER");
-        if (resolvedDriver.contains("postgresql")) {
-            return Optional.of(QuarkusPostgreSQL95Dialect.class.getName());
+        String resolvedKind = kind.orElse("NO_KIND");
+        if (DatabaseKind.isPostgreSQL(resolvedKind)) {
+            return Optional.of(QuarkusPostgreSQL10Dialect.class.getName());
         }
-        if (resolvedDriver.contains("org.h2.Driver")) {
+        if (DatabaseKind.isH2(resolvedKind)) {
             return Optional.of(QuarkusH2Dialect.class.getName());
         }
-        if (resolvedDriver.contains("org.mariadb.jdbc.Driver")) {
+        if (DatabaseKind.isMariaDB(resolvedKind)) {
             return Optional.of(MariaDB103Dialect.class.getName());
         }
-
-        if (resolvedDriver.contains("com.mysql.cj.jdbc.Driver")) {
+        if (DatabaseKind.isMySQL(resolvedKind)) {
             return Optional.of(MySQL8Dialect.class.getName());
         }
-        if (resolvedDriver.contains("org.apache.derby.jdbc.ClientDriver")) {
+        if (DatabaseKind.isDerby(resolvedKind)) {
             return Optional.of((DerbyTenSevenDialect.class.getName()));
         }
+        if (DatabaseKind.isMsSQL(resolvedKind)) {
+            return Optional.of((SQLServer2012Dialect.class.getName()));
+        }
 
-        String error = driver.isPresent()
-                ? "Hibernate extension could not guess the dialect from the driver '" + resolvedDriver
+        String error = kind.isPresent()
+                ? "Hibernate extension could not guess the dialect from the kind '" + resolvedKind
```

Comment:
```suggestion
                ? "Hibernate extension could not guess the dialect from the database kind '" + resolvedKind
```

Reply:
Done and fixed a few other things in this message too.